apsuite/commissioning_scripts/meas_coupling_tune.py

```diff
@@ -0,0 +1,169 @@
+
+    # --- fitting parameters ---
+
+    @property
+    def coeff1(self):
+        """."""
+        return self.position[0]
+
+    @coeff1.setter
+    def coeff1(self, value):
+        """."""
+        self.position[0] = value
+
+    @property
+    def offset1(self):
+        """."""
+        return self.position[1]
+
+    @offset1.setter
+    def offset1(self, value):
+        """."""
+        self.position[1] = value
+
+    @property
+    def coeff2(self):
+        """."""
+        return self.position[2]
+
+    @coeff2.setter
+    def coeff2(self, value):
+        """."""
+        self.position[2] = value
+
+    @property
+    def offset2(self):
+        """."""
+        return self.position[3]
+
+    @offset2.setter
+    def offset2(self, value):
+        """."""
+        self.position[3] = value
+
+    @property
+    def coupling(self):
+        """."""
+        return self.position[4]
+
+    @coupling.setter
+    def coupling(self, value):
+        """."""
+        self.position[4] = value
+
+    def calc_obj_fun(self):
+        """."""
+        # print(self.position)
+        coeff1, offset1, coeff2, offset2, coupling = self.position
+        _, tune1, tune2 = \
+            FitTunes.calc_tunes(
+                self._param, coeff1, offset1, coeff2, offset2, coupling)
+        residue = sum((tune1 - self._tune1)**2 + (tune2 - self._tune2)**2)
+        return residue
+
+    def initialization(self):
+        """."""
+        self._calc_init_parms()
+        coeff1, offset1, coeff2, offset2, _ = self.position
+        self.deltas = _np.array([
+            0.002 * (coeff1 - coeff2),
+            0.002 * (offset1 - offset2),
+            0.002 * (coeff1 - coeff2),
+            0.002 * (offset1 - offset2),
+            0.5 * 0.1/100])
+
+    @staticmethod
+    def calc_tunes(
+            param=None,
+            coeff1=None, offset1=None,
+            coeff2=None, offset2=None, coupling=None):
+        """."""
+        fx_ = coeff1 * param + offset1
+        fy_ = coeff2 * param + offset2
+        tune1, tune2 = _np.zeros(param.shape), _np.zeros(param.shape)
+        for i in range(len(param)):
+            mat = _np.array([[fx_[i], coupling/2], [coupling/2, fy_[i]]])
+            eigvalues, _ = _np.linalg.eig(mat)
+            tune1[i], tune2[i] = eigvalues
+            if tune1[i] <= tune2[i]:
+                tune1[i], tune2[i] = tune2[i], tune1[i]
+        return param, tune1, tune2
+
+    def fitting_plot(self, title=None, fig=None):
+        """."""
+        position = self.position
+        if title is None:
+            title = 'Transverse Linear Coupling: {:.2f} %'.format(
+                position[-1] * 100)
+        prop_cycle = _plt.rcParams['axes.prop_cycle']
+        colors = prop_cycle.by_key()['color']
+        if fig is None:
+            fig = _plt.figure()
+        param, fittune1, fittune2 = FitTunes.calc_tunes(self._param, *position)
+
+        # plot meas data
+        _plt.plot(
```

Comment:
If you use `matplotlib.gridspec` you will be able to control the size and position of the `axes` inside the figure, improving visualization.

Reply:
I implemented a _back-of-the-envelope_ plot method so that I could see the fitting results. You guys are much more familiar with _matplotlib_. Do you wan to commit a generalized plot method? (for example, one that allows users to choose where to plot).